File: src/main/java/no/nav/syfo/util/Toggle.java
package no.nav.syfo.util;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class Toggle {

    private boolean toggleBatch;
    private boolean toggleBatchSak;
    private String envName;

    public Toggle(
            @Value("${toggle.enable.batch:false}") boolean toggleBatch,
            @Value("${toggle.enable.batch.sak:true}") boolean toggleBatchSak,
            @Value("${nais.cluster.name:prod-fss}") String envName
    ) {
        this.toggleBatch = toggleBatch;
        this.toggleBatchSak = toggleBatchSak;
        this.envName = envName;
    }

    public boolean erPreprod() {
        return envName.contains("dev-fss");
    }
}

File: src/main/kotlin/no/nav/syfo/toggle/Toggle.kt
package no.nav.syfo.toggle

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class Toggle(
    @Value("\${toggle.enable.batch:false}") private val toggleBatch: Boolean,
    @Value("\${toggle.enable.batch.sak:true}") private val toggleBatchSak: Boolean,
    @Value("\${nais.cluster.name:prod-fss}") private val envName: String
) {
    fun erPreprod(): Boolean {
        return envName.contains("dev-fss")
    }
}
